addons/mail/static/tests/thread/attachment_list_tests.js

```diff
@@ -97,6 +97,40 @@ QUnit.test(
     }
 );
 
+QUnit.test(
+    "link-type attachment should have open button instead of download button",
+    async (assert) => {
+        const pyEnv = await startServer();
+        const channelId = pyEnv["discuss.channel"].create({
+            channel_type: "channel",
+            name: "channel1",
+        });
+        const messageAttachmentId = pyEnv["ir.attachment"].create({
+            name: "url.example",
+            mimetype: "text/plain",
+            type: "url",
+            url: "https://www.odoo.com",
+        });
+        pyEnv["mail.message"].create({
+            attachment_ids: [messageAttachmentId],
+            body: "<p>Test</p>",
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        });
+        const { openDiscuss } = await start();
+        openDiscuss(channelId);
+        await contains(".o-mail-AttachmentCard");
+        await contains(".o-mail-AttachmentCard-aside a[title='Open Link']");
+        await contains(".o-mail-AttachmentCard-aside button[title='Download']", { count: 0 });
+        assert.strictEqual(
+            document.querySelector(`.o-mail-AttachmentCard-aside a[title='Open Link']`).target,
+            "_blank",
+            "attachment should have a open link button in a new tab"
```

Comment:
The showing of buttons "Download"/"Remove" have been changed into a dropdown menu, starting from 17.0:
https://github.com/odoo/odoo/pull/185219

I know this test is attachment card while the improvement / fix above affects only images, but I want to point our that checking `count: 0` without an equivalent positive assertion using similar selector is very weak to such change.

I would strongly advice to craft 2 attachments cards in test, one is URL and the other is not. That way we can use same selectors in positive and negative assertion, thus the negative assertion is much robust.